charts/opentelemetry-demo/templates/_pod.tpl

```diff
@@ -41,25 +41,33 @@ Get Services Port Mapping
 {{- end }}
 
 {{/*
-Get Pod Env 
+Get Pod Env
 */}}
 {{- define "otel-demo.pod.env" -}}
 {{- $prefix := include "otel-demo.name" $ }}
 
+{{- if .default.enabled  }}
+{{- toYaml .default.env }}
+{{- end }}
+
 {{- if .env }}
-{{- toYaml .env }}
+{{ toYaml .env }}
 {{- end }}
 
-{{- if .observability.otelcol.enabled }} 
+{{- if .observability.otelcol.enabled }}
 - name: OTEL_EXPORTER_OTLP_ENDPOINT
   value: http://{{ include "otel-demo.name" . }}-otelcol:4317
 - name: OTEL_RESOURCE_ATTRIBUTES
+{{- if .default.enabled  }}
+  value: service.name={{ .name | kebabcase }},k8s.namespace.name=$(OTEL_K8S_NAMESPACE),k8s.node.name=$(OTEL_RESOURCE_ATTRIBUTES_NODE_NAME),k8s.pod.name=$(OTEL_RESOURCE_ATTRIBUTES_POD_NAME)
```

Comment:
I think this line needs to be configurable.  These should be enabled by default but should be able to be removed.  probably another section in the `default` object in the values.yaml.

Reply:
you means this?
```
default:
  enabled: true
  env:
    - name: OTEL_K8S_NAMESPACE
      enabled: true
      valueFrom:
        fieldRef:
          apiVersion: v1
          fieldPath: metadata.namespace
    - name: OTEL_RESOURCE_ATTRIBUTES_POD_NAME
      enabled: false
      valueFrom:
        fieldRef:
          apiVersion: v1
          fieldPath: metadata.name
```

and finally `OTEL_RESOURCE_ATTRIBUTES` would like:
```
- name: OTEL_RESOURCE_ATTRIBUTES
  value: service.name=cart-service,k8s.namespace.name=$(OTEL_K8S_NAMESPACE)
```

How about use a prefix, like OTEL_RESOURCE, to mark which envs shoud add into the `OTEL_RESOURCE_ATTRIBUTES`?

Reply:
Actually, "OTEL_RESOURCE_ATTRIBUTES" is an environment variable right? Can we put it directly in default?

Reply:
ok, put it directly in default section is esay and Flexible way.

Reply:
I like it, but since `OTEL_RESOURCE_ATTRIBUTES` relies on other env vars, it definitely has to be after the others in the rendered yaml file.  I'm pretty sure `toYaml` won't preserve the order of a dictionary, so there isn't a guarantee that it says at the end of the list like in the values.yaml.  Can you test that?  Sorry, I didn't think about it till this morning.

Reply:
You. might be able to do something fancy with the [`omit` function](https://helm.sh/docs/chart_template_guide/function_list/#omit) and the [`get` function](https://helm.sh/docs/chart_template_guide/function_list/#get) to omit `OTEL_RESOURCE_ATTRIBUTES` from the initial `toYaml` of the dictionary and then do a specific `get` for `OTEL_RESOURCE_ATTRIBUTES` after.

Reply:
I will change the logic to guarantee it.